code/datums/martial.dm

```diff
@@ -104,24 +92,18 @@
 
 	var/atk_verb = pick("left hook","right hook","straight punch")
 
-	var/damage = rand(5,8)
-	if(A.dna)
-		damage += A.dna.species.punchmod
+	var/damage = rand(5,8) + A.dna.species.punchmod
 	if(!damage)
-		if(A.dna)
-			playsound(D.loc, A.dna.species.miss_sound, 25, 1, -1)
-		else
-			playsound(D.loc, 'sound/weapons/punchmiss.ogg', 25, 1, -1)
-			D.visible_message("<span class='warning'>[A] has attempted to hit [D] with a [atk_verb]!</span>")
-			add_logs(A, D, "attempted to hit", atk_verb)
-			return 0
+		playsound(D.loc, A.dna.species.miss_sound, 25, 1, -1)
+		D.visible_message("<span class='warning'>[A] has attempted to hit [D] with a [atk_verb]!</span>")
+		add_logs(A, D, "attempted to hit", atk_verb)
+		return 0
 
 
 	var/obj/item/organ/limb/affecting = D.get_organ(ran_zone(A.zone_sel.selecting))
 	var/armor_block = D.run_armor_check(affecting, "melee")
 
-	playsound(D.loc, 'sound/weapons/punch1.ogg', 25, 1, -1)
-
+	playsound(D.loc, A.dna.species.attack_sound, 25, 1, -1)
 
```

Comment:
Fixed a lot of copypasta and simplified the code because the dna checks are no longer needed.